src/benchmark/operators/table_scan_sorted_benchmark.cpp

```diff
@@ -0,0 +1,277 @@
+      str = std::string(string_length - str.length(), '0').append(str);
+      table->append({str});
+    }
+  }
+
+  if (encoding_type != EncodingType::Unencoded) {
+    ChunkEncoder::encode_all_chunks(table, SegmentEncodingSpec(encoding_type));
+  }
+
+  if (order_by.has_value()) {
+    for (auto& chunk : table->chunks()) {
+      chunk->get_segment(ColumnID(0))->set_sort_order(order_by.value());
+    }
+  }
+
+  table_wrapper = std::make_shared<TableWrapper>(std::move(table));
+  table_wrapper->execute();
+
+  return table_wrapper;
+}
+
+void BM_TableScanSorted(
+    benchmark::State& state, const int table_size, const double selectivity,
+    const PredicateCondition predicate_condition, const EncodingType encoding_type,
+    const std::optional<OrderByMode> order_by,
+    std::function<std::shared_ptr<TableWrapper>(const int, const EncodingType, const OrderByMode)> table_creator) {
+  _clear_cache();
+
+  AllTypeVariant search_value;
+  switch (predicate_condition) {
+    case PredicateCondition::LessThanEquals:
+    case PredicateCondition::LessThan:
+      search_value = static_cast<int>(table_size * selectivity);
+      break;
+    case PredicateCondition::GreaterThan:
+    case PredicateCondition::GreaterThanEquals:
+      search_value = static_cast<int>(table_size - table_size * selectivity);
+      break;
+    default:
+      // TODO(cmfcmf) Improve logic for other predicates.
+      Fail("Unsupported predicate condition");
+  }
+  const auto table_wrapper = table_creator(table_size, encoding_type, order_by.value_or(OrderByMode::Ascending));
+  const auto table_column_definitions = table_wrapper->get_output()->column_definitions();
+
+  const auto column_index = ColumnID(0);
+
+  const auto column_definition = table_column_definitions.at(column_index);
+
+  if (column_definition.data_type == DataType::String) {
+    const auto str = std::to_string(type_cast_variant<int>(search_value));
+    search_value = std::string(string_size - str.length(), '0').append(str);
+  }
+
+  const auto column_expression =
+      pqp_column_(column_index, column_definition.data_type, column_definition.nullable, column_definition.name);
+
+  auto predicate =
+      std::make_shared<BinaryPredicateExpression>(predicate_condition, column_expression, value_(search_value));
+
+  auto warm_up = std::make_shared<TableScan>(table_wrapper, predicate);
+  warm_up->execute();
+  for (auto _ : state) {
+    auto table_scan = std::make_shared<TableScan>(table_wrapper, predicate);
+    table_scan->execute();
+  }
+}
+
+std::shared_ptr<TableWrapper> sorted_int_table(const int table_size, const EncodingType encoding_type,
+                                               const OrderByMode order_by) {
+  return create_int_table(table_size, encoding_type, std::make_optional(order_by));
+}
+
+std::shared_ptr<TableWrapper> unsorted_int_table(const int table_size, const EncodingType encoding_type,
+                                                 const OrderByMode order_by) {
```

Comment:
I guess you need the order_by for the function signature?
I don't know, if it would be better to use an optional?
You could at least remove the variable name.